control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1396,6 +1396,26 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context) (bool,
 	if len(errs) > 0 {
 		return false, fmt.Errorf("failed to delete ingress controllers: %w", errors.NewAggregate(errs))
 	}
+
+	// force deleting pods under openshift-ingress to unblock ingress-controller deletion
+	// ingress-operator is dependent on these pods deletion on removing the finalizers of ingress-controller
+	routerPods := &corev1.PodList{}
+	if err := r.uncachedClient.List(ctx, routerPods, &client.ListOptions{Namespace: "openshift-ingress"}); err != nil {
+		return false, fmt.Errorf("failed to list pods under openshift-ingress namespace: %w", err)
+	}
```

Comment:
would it make sense to just delete the namespace?

Reply:
Deleting namespace is not helping, since its also stuck because of this pod is stuck in terminating state. Seeing this condition on namespace describe `NamespaceContentRemaining                    True    Thu, 22 Sep 2022 18:12:18 +0530  SomeResourcesRemain     Some resources are remaining: pods. has 1 resource instances` If namespace alone deleted.

Reply:
@dharaneeshvrd I'm confused, how does deleting the pod directly (instead of the namespace) helps If that pod deletion is stuck?

Reply:
I have attached the screenshot below where on its right side, I am trying to force delete namespace and its stuck. On the other side, described the namespace where the conditions section has one of the condition as **_NamespaceContentRemaining_** , I have highlighted that.

Please take a look @enxebre 

![image](https://user-images.githubusercontent.com/17947543/194231198-2c339c81-a13b-4e44-9e3a-ada63423f597.png)

Pod is a child resource of namespace and namespace is waiting for all its child resources to get deleted before it proceeds to its deletion. That's why wanted to force delete the pod to unblock namespace deletion which unblocks ingress controller deletion.

Discussed about this issue and fix here https://coreos.slack.com/archives/C01C8502FMM/p1661178025483469

Reply:
when you signal the namespace for deletion, doesn't the pod get a deletion timestamp as well?

Reply:
@enxebre pod already got a deletion timestamp by ingress-controller. At that time namespace did n't get have deletion timestamp. Deleting the namespace is not affecting anything on the pod.